File: Readio/src/main/java/ninegle/Readio/library/domain/Library.java
package ninegle.Readio.library.domain;

import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.*;
import org.hibernate.annotations.UpdateTimestamp;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import ninegle.Readio.user.domain.User;

@Table(name = "`library`")
@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Library {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "user_id")
	private User user;

	private String libraryName;

	//나 자신 즉 library 삭제하면 libraryBook도 삭제
	@OneToMany(mappedBy = "library", cascade = CascadeType.ALL, orphanRemoval = true)
	private List<LibraryBook> libraryBook;

	private LocalDateTime createdAt;

	@UpdateTimestamp
	private LocalDateTime updatedAt;

	public Library(String libraryname, User user) {
		this.createdAt = LocalDateTime.now();
		this.libraryName = libraryname;
		this.user = user;
	}

	public Library changeLibraryName(String newLibraryName) {
		this.libraryName = newLibraryName;
		return this;
	}

}

File: Readio/src/main/java/ninegle/Readio/library/domain/Library.kt
package ninegle.Readio.library.domain

import jakarta.persistence.*
import ninegle.Readio.user.domain.User
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime

@Table(name = "`library`")
@Entity
class Library(

    @field: Column(nullable = false)
    var libraryName: String,

    @field:JoinColumn(name = "user_id")
    @field:ManyToOne (fetch = FetchType.LAZY)
    var user: User)
{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    //나 자신 즉 library 삭제하면 libraryBook도 삭제
    @OneToMany(mappedBy = "library", cascade = [CascadeType.ALL], orphanRemoval = true)
    val libraryBook: List<LibraryBook> = mutableListOf()

    var createdAt: LocalDateTime = LocalDateTime.now()

    @UpdateTimestamp
    var updatedAt: LocalDateTime? = null

    fun changeLibraryName(newLibraryName: String): Library {
        this.libraryName = newLibraryName
        return this
    }
}
